teamcity: failed test: TestManyChangefeedsOneTable

The following tests appear to have failed on master (testrace): TestManyChangefeedsOneTable/sinkless, TestManyChangefeedsOneTable

You may want to check [for open issues](https://github.com/cockroachdb/cockroach/issues?q=is%3Aissue+is%3Aopen+TestManyChangefeedsOneTable).

[#1099042](https://teamcity.cockroachdb.com/viewLog.html?buildId=1099042):


```
TestManyChangefeedsOneTable
--- FAIL: testrace/TestManyChangefeedsOneTable (0.000s)
Test ended in panic.




TestManyChangefeedsOneTable
--- FAIL: test/TestManyChangefeedsOneTable (1.440s)
Test ended in panic.




TestManyChangefeedsOneTable/sinkless
...ler).worker(0xc001b6e2d0, 0x4f27d80, 0xc001068210)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:196 +0xa8
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2(0x4f27d80, 0xc001068210)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:165 +0x56
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc00226f650, 0xc000d5d7a0, 0xc00226f640)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:198 +0x146
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:191 +0xc4

goroutine 23322 [chan receive, 41 minutes]:
github.com/cockroachdb/cockroach/pkg/util/log.(*loggingT).gcDaemon(0xc00186a780)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:1274 +0x8d
created by github.com/cockroachdb/cockroach/pkg/util/log.NewSecondaryLogger
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/secondary_log.go:81 +0x4c5

goroutine 33694 [sync.Cond.Wait]:
sync.runtime_notifyListWait(0xc00134e150, 0xc000001046)
	/usr/local/go/src/runtime/sema.go:510 +0xeb
sync.(*Cond).Wait(0xc00134e140)
	/usr/local/go/src/sync/cond.go:56 +0x8e
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc001b6e2d0, 0x4f27d80, 0xc002321860)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:196 +0xa8
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2(0x4f27d80, 0xc002321860)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:165 +0x56
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc00226f450, 0xc000d5d7a0, 0xc00226f440)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:198 +0x146
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:191 +0xc4

goroutine 23321 [chan receive, 41 minutes]:
github.com/cockroachdb/cockroach/pkg/util/log.(*loggingT).gcDaemon(0xc00186a640)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:1274 +0x8d
created by github.com/cockroachdb/cockroach/pkg/util/log.NewSecondaryLogger
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/secondary_log.go:81 +0x4c5

goroutine 33747 [select]:
github.com/cockroachdb/cockroach/pkg/storage.(*Store).startGossip.func4(0x4f27d80, 0xc001069680)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:1477 +0x443
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc00226fb20, 0xc000d5d7a0, 0xc000a76410)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:198 +0x146
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:191 +0xc4

goroutine 33564 [chan receive, 40 minutes]:
github.com/cockroachdb/cockroach/pkg/util/netutil.MakeServer.func2(0x4f27d80, 0xc002898210)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/netutil/net.go:99 +0x7d
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc002200f90, 0xc000d5d7a0, 0xc001539e80)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:198 +0x146
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:191 +0xc4

goroutine 33781 [select]:
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).Start.func2(0x4f27d80, 0xc0015ce8d0)
	/go/src/github.com/cockroachdb/cockroach/pkg/jobs/registry.go:299 +0x141
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc00226fd60, 0xc000d5d7a0, 0xc002321e90)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:198 +0x146
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:191 +0xc4




```

Please assign, take a look and update the issue accordingly.

--- a/pkg/ccl/changefeedccl/changefeed_processors.go
+++ b/pkg/ccl/changefeedccl/changefeed_processors.go
@@ -166,8 +166,9 @@ func (ca *changeAggregator) Start(ctx context.Context) context.Context {
 	// Give errCh enough buffer both possible errors from supporting goroutines,
 	// but only the first one is ever used.
 	ca.errCh = make(chan error, 2)
-	ca.pollerDoneCh = make(chan struct{})
+
 	if err := ca.flowCtx.Stopper().RunAsyncTask(ctx, "changefeed-poller", func(ctx context.Context) {
+		ca.pollerDoneCh = make(chan struct{})
 		defer close(ca.pollerDoneCh)
 		var err error
 		if storage.RangefeedEnabled.Get(&ca.flowCtx.Settings.SV) {

--- a/pkg/ccl/changefeedccl/changefeed_test.go
+++ b/pkg/ccl/changefeedccl/changefeed_test.go
@@ -1521,7 +1521,7 @@ func TestManyChangefeedsOneTable(t *testing.T) {
 	testFn := func(t *testing.T, db *gosql.DB, f testfeedFactory) {
 		sqlDB := sqlutils.MakeSQLRunner(db)
 		sqlDB.Exec(t, `CREATE TABLE foo (a INT PRIMARY KEY, b STRING)`)
-		sqlDB.Exec(t, `INSERT INTO foo VALUES (0, 'v0')`)
+		sqlDB.Exec(t, `INSERT INTO foo VALUES (0, 'init')`)
 
 		foo1 := f.Feed(t, `CREATE CHANGEFEED FOR foo`)
 		defer foo1.Close(t)
@@ -1530,6 +1530,12 @@ func TestManyChangefeedsOneTable(t *testing.T) {
 		foo3 := f.Feed(t, `CREATE CHANGEFEED FOR foo`)
 		defer foo3.Close(t)
 
+		// Make sure all the changefeeds are going.
+		assertPayloads(t, foo1, []string{`foo: [0]->{"a": 0, "b": "init"}`})
+		assertPayloads(t, foo2, []string{`foo: [0]->{"a": 0, "b": "init"}`})
+		assertPayloads(t, foo3, []string{`foo: [0]->{"a": 0, "b": "init"}`})
+
+		sqlDB.Exec(t, `UPSERT INTO foo VALUES (0, 'v0')`)
 		assertPayloads(t, foo1, []string{
 			`foo: [0]->{"a": 0, "b": "v0"}`,
 		})

--- a/pkg/ccl/changefeedccl/changefeed_test.go
+++ b/pkg/ccl/changefeedccl/changefeed_test.go
@@ -367,10 +367,12 @@ func TestChangefeedSchemaChangeNoBackfill(t *testing.T) {
 			sqlDB.Exec(t, `INSERT INTO add_column VALUES (1)`)
 			addColumn := f.Feed(t, `CREATE CHANGEFEED FOR add_column`)
 			defer addColumn.Close(t)
+			assertPayloads(t, addColumn, []string{
+				`add_column: [1]->{"a": 1}`,
+			})
 			sqlDB.Exec(t, `ALTER TABLE add_column ADD COLUMN b STRING`)
 			sqlDB.Exec(t, `INSERT INTO add_column VALUES (2, '2')`)
 			assertPayloads(t, addColumn, []string{
-				`add_column: [1]->{"a": 1}`,
 				`add_column: [2]->{"a": 2, "b": "2"}`,
 			})
 		})

